### PR TITLE
add username parameter to dump-keyfamily CORE-3039

### PR DESCRIPTION
**Pull request by mmaxim**
@oconnor663 ?

### CODE REVIEW

File: go/client/cmd_dump_keyfamily.go [hunk 1]
```diff
@@ -23,12 +23,27 @@ func indentSpace(level int) string {
 
 type CmdDumpKeyfamily struct {
 	libkb.Contextified
+	user string
+}
+
+func (v *CmdDumpKeyfamily) ParseArgv(ctx *cli.Context) error {
+	nargs := len(ctx.Args())
+	if nargs > 1 {
+		return fmt.Errorf("dump-keyfamily only takes on argument, the user to lookup")
```
Comment (oconnor663) [inline go/client/cmd_dump_keyfamily.go]: `s/on/one/`

File: go/client/cmd_dump_keyfamily.go [hunk 2]
```diff
@@ -23,12 +23,27 @@ func indentSpace(level int) string {
...
 }
 
 func NewCmdDumpKeyfamily(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
 	return cli.Command{
-		Name:  "dump-keyfamily",
-		Flags: []cli.Flag{},
+		Name:         "dump-keyfamily",
+		ArgumentHelp: "[username]",
+		Usage:        "Print out a user's current key family",
+		Description:  "Print out a user's current key family. Don't specify a username to dump out your own keys.",
```
Comment (oconnor663) [inline go/client/cmd_dump_keyfamily.go]: In my head "Don't specify a username..." sounds like it's a warning :) Maybe "with no argument, prints your own key family."

Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: Yeah I agree, I just copied the phrasing from the id command.

File: go/client/cmd_dump_keyfamily.go [hunk 3]
```diff
@@ -48,21 +63,31 @@ func (v *CmdDumpKeyfamily) Run() (err error) {
 	if !currentStatus.LoggedIn {
 		return fmt.Errorf("Not logged in.")
 	}
-	myUID := currentStatus.User.Uid
+
+	var UID keybase1.UID
+	if v.user != "" {
+		res := v.G().Resolver.Resolve(v.user)
```
Comment (oconnor663) [inline go/client/cmd_dump_keyfamily.go]: Using `Resolver` means we're trusting the server to return the correct UID to us. Is that bad? Maybe the server could swap in a UID for a user with a suspiciously similar username that was under the server's control. I'm just being paranoid, but if it's easy to load user directly by username, it might be marginally better to do it that way?

**Comment by oconnor663** [comment]
LGTM, with the one question about username vs uid. @maxtaco do you care about that one?

File: go/client/cmd_dump_keyfamily.go (hunk 3 above)
Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: @mmaxim why did you need to call this explicitly? i believe the resolver keeps you honest w/r/t uid<->username mapping.

Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: I'd have to add a new RPC for that, do you think that is the better play?

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: BTW, what I said is true only if you have `LoadUser` doing the resolve for you...

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: (So isn't true for this code).

Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: The `LoadUser` RPC only takes UID, so I think I need to add a new one that takes username instead so I can pass it to `libkb.LoadUser`. Does that sound right?

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: I think so for the reasons that jack gives.  Maybe just support the Username parameter to LoadUserArg when calling LoadUSer.

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: yes it sounds right...

Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: Ok I'll make that change, I'll also need to do that for `keybase.LoadPublicKeys`

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: you'll get the UID back as a result of the first RPC so you won't need to change the second RPC.

**Comment by maxtaco** [comment]
there's a core problem here which is that this command isn't implemented with the logic in the service. @oconnor663 why is this one a bit of an exception?

File: go/client/cmd_dump_keyfamily.go (hunk 3 above)
Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: I might as well add it though no? Might be useful someday.

Comment (maxtaco) [inline go/client/cmd_dump_keyfamily.go]: i would argue against, i think it's going to be a lot of bloat!

Comment (mmaxim) [inline go/client/cmd_dump_keyfamily.go]: Sounds good!

File: go/service/user.go [hunk 4]
```diff
@@ -99,6 +99,16 @@ func (h *UserHandler) LoadUser(_ context.Context, arg keybase1.LoadUserArg) (use
 	return
 }
 
+func (h *UserHandler) LoadUserByName(_ context.Context, arg keybase1.LoadUserByNameArg) (user keybase1.User, err error) {
+	u, err := libkb.LoadUser(libkb.LoadUserArg{Name: arg.Username, Contextified: libkb.NewContextified(h.G())})
```
Comment (patrickxb) [inline go/service/user.go]: can you use libkb.NewLoadUserByNameArg()?

Comment (mmaxim) [inline go/service/user.go]: Yes, I just copy/pasted what was in `LoadUser` though.

Comment (mmaxim) [inline go/service/user.go]: Fixed.

Comment (patrickxb) [inline go/service/user.go]: Sorry, probably an oversight on my part when I added NewLoadUser*Arg() funcs.

**Comment by maxtaco** [comment]
👍
